Source/WebCore/css/typedom/InlineStylePropertyMap.cpp

```diff
@@ -105,6 +107,8 @@ bool InlineStylePropertyMap::setProperty(CSSPropertyID propertyID, Ref<CSSValue>
     bool didFailParsing = false;
     bool important = false;
     m_element->setInlineStyleProperty(propertyID, value->cssText(), important, &didFailParsing);
+    if (!didFailParsing)
+        m_cachedProperties.set(propertyID, WTFMove(value));
```

Comment:
Not sure I understand this patch. Here we serialize and rotate the CSSValue through the parser and lose something in the process? Why don't we just set the inline style to the actual CSSValue provided?

Reply:
I uploaded an alternate proposal here https://github.com/WebKit/WebKit/pull/7519 to show you the work involved in making this approach work.